### PR TITLE
Issue #1776 Moving default Origin version to v3.7.1

### LINKED ISSUE #1776: Update default OpenShift Origin version to 3.7.1 in Minishift
Now that Origin v3.7.0 is released we should move to the latest GA version.

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Fix #1776

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
retest this please

**Comment by LalatenduMohanty** [comment]
Last integration tests failed with following errors
```
12:40:30   Scenario: User can deploy a Ruby example application                                                                                # features/basic.feature:141
12:40:30     Given Minishift has state "Running"                                                                                               # integration_test.go:162 -> *Minishift.shouldHaveState
12:40:32     When executing "oc login --username=developer --password=developer" succeeds                                                      # integration_test.go:722 -> executingOcCommandSucceedsOrFails
12:40:34     And executing "oc new-project ruby" succeeds                                                                                      # integration_test.go:722 -> executingOcCommandSucceedsOrFails
12:40:37     And executing "oc new-app centos/ruby-22-centos7~https://github.com/openshift/ruby-ex.git" succeeds                               # integration_test.go:722 -> executingOcCommandSucceedsOrFails
12:43:10     And services "ruby-ex" rollout successfully                                                                                       # integration_test.go:188 -> *Minishift.rolloutServicesSuccessfully
12:43:12     And executing "oc expose svc/ruby-ex" succeeds                                                                                    # integration_test.go:722 -> executingOcCommandSucceedsOrFails
12:43:14     Then status code of HTTP request to "/" of service "ruby-ex" in namespace "ruby" is equal to "200"                                # integration_test.go:795 -> getRoutingUrlAndVerifyHTTPResponse
12:43:14     status code of response from http://ruby-ex-ruby.192.168.42.201.nip.io is not equal to expected string. Expected: 200, Actual: 503
```

**Comment by LalatenduMohanty** [comment]
Got the same errors from the 2nd run https://ci.centos.org/job/minishift-pr/2993/consoleFull

**Comment by LalatenduMohanty** [comment]
retest this please

**Comment by gbraad** [comment]
Any news on the status of this change? Do the sporadic errors still occur, and if so, do we already know what causes this?

**Comment by budhrg** [comment]
retest this please

**Comment by budhrg** [comment]
retest this please

**Comment by agajdosi** [comment]
The issue is probably what we see in the downstream, I am trying to resolve it in #1838. The fix is to delay the http request check a little bit or to add a retry logic.

**Comment by LalatenduMohanty** [comment]
As seen by @budhrg in ci.centos.org, `minishift start` seems to fail with error `Persistent volume creation failed`. It might be caused someway by the `anyuid` add-on being enabled during the start.

**Comment by hferentschik** [comment]
> The issue is probably what we see in the downstream, I am trying to resolve it in #1838. The fix is to delay the http request check a little bit or to add a retry logic.

-1 for delay and retry. This should not be required and if it is we are running into a bug.

**Comment by LalatenduMohanty** [comment]
retest this please

**Comment by LalatenduMohanty** [comment]
retest this please.

**Comment by LalatenduMohanty** [comment]
After merging #1871 only one test is failing, refer https://ci.centos.org/job/minishift-pr/3155/console
```
# integration_test.go:196 -> *Minishift.rolloutServicesSuccessfully
10:26:11     And executing "oc expose svc/ruby-ex" succeeds                                                                                    # integration_test.go:695 -> executingOcCommandSucceedsOrFails
10:26:11     Then status code of HTTP request to "/" of service "ruby-ex" in namespace "ruby" is equal to "200"                                # integration_test.go:768 -> getRoutingUrlAndVerifyHTTPResponse
10:26:11     status code of response from http://ruby-ex-ruby.192.168.42.76.nip.io is not equal to expected string. Expected: 200, Actual: 503
```

**Comment by LalatenduMohanty** [comment]
retest this please

**Comment by agajdosi** [comment]
I am seeing the problem with nodejs-ex in downstream also:
```
07:00:00   Scenario: User deploys nodejs example application from OpenShift repository                 # features/cmd-openshift.feature:35
07:00:00     When executing "oc new-app https://github.com/openshift/nodejs-ex -l name=myapp" succeeds # integration_test.go:709 -> executingOcCommandSucceedsOrFails
07:00:00     Command 'new-app https://github.com/openshift/nodejs-ex -l name=myapp' did not execute successfully. cmdExit: 1, cmdErr: error: The connection to the server 192.168.42.75:8443 was refused - did you specify the right host or port?
07:00:00 error: The connection to the server 192.168.42.75:8443 was refused - did you specify the right host or port?
07:00:00 error: The connection to the server 192.168.42.75:8443 was refused - did you specify the right host or port?
07:00:00 error: The connection to the server 192.168.42.75:8443 was refused - did you specify the right host or port?
07:00:00 error: can't lookup images: Post https://192.168.42.75:8443/apis/image.openshift.io/v1/namespaces/myproject/imagestreamimports: dial tcp 192.168.42.75:8443: getsockopt: connection refused
07:00:00 error: no match for "nodejs"
07:00:00 
07:00:00 The 'oc new-app' command will match arguments to the following types:
07:00:00 
07:00:00   1. Images tagged into image streams in the current project or the 'openshift' project
07:00:00      - if you don't specify a tag, we'll add ':latest'
07:00:00   2. Images in the Docker Hub, on remote registries, or on the local Docker engine
07:00:00   3. Templates in the current project or the 'openshift' project
07:00:00   4. Git repository URLs or local paths that point to Git repositories
07:00:00 
07:00:00 --allow-missing-images can be used to point to an image that does not exist yet.
07:00:00 
07:00:00 See 'oc new-app -h' for examples.
07:00:00 
```

I am also surprised that no retries are being reported when checking http request to ruby-ex application.

**Comment by budhrg** [comment]
@LalatenduMohanty , are we still having issue in updating the version here?

**Comment by LalatenduMohanty** [comment]
> @LalatenduMohanty , are we still having issue in updating the version here?
Yes, I still failures, See https://ci.centos.org/job/minishift-pr/3369/

**Comment by LalatenduMohanty** [comment]
retest this please

**Comment by LalatenduMohanty** [comment]
@budhrg I commented the below test and it passed https://ci.centos.org/job/minishift-pr/3409/. Will test with adding a sleep between this test and the next test i.r. ruby example deployment
```
-  Scenario: Restarting the OpenShift cluster
-  Note: This step is based on observation and might be unstable in some environments. It checks for the time when container
-        finished last time. When container is new and had never finished then this time value is set to 0001-01-01T00:00:00Z.
-        On restart of OpenShift cluster containers are terminated, which sets FinishedAt to actual time. This value persist
-        after next start of container.
-     Given stdout of command "minishift ssh -- "docker inspect --format={{.State.FinishedAt}} origin"" is equal to "0001-01-01T00:00:00Z"
-      When executing "minishift openshift restart" succeeds
-      Then stdout should contain "Restarting OpenShift"
-       And stdout of command "minishift ssh -- "docker inspect --format={{.State.FinishedAt}} origin"" is not equal to "0001-01-01T00:00:00Z"
-
```

**Comment by LalatenduMohanty** [comment]
Just to be clear, this issue is not reproducible when I tried manually. i.e. 

* minishift openshift restart
* oc new-app https://github.com/openshift/nodejs-ex -l name=myapp

File: test/integration/features/cmd-openshift.feature [hunk 1]
```diff
@@ -33,10 +33,12 @@ cluster in VM provided by Minishift.
      Given stdout of command "minishift ssh -- "docker inspect --format={{.State.FinishedAt}} origin"" is equal to "0001-01-01T00:00:00Z"
       When executing "minishift openshift restart" succeeds
       Then stdout should contain "Restarting OpenShift"
+       And user waits "5" seconds
```
Comment (budhrg) [inline test/integration/features/cmd-openshift.feature]: @LalatenduMohanty  This doesn't need then

File: test/integration/features/cmd-openshift.feature [hunk 2]
```diff
@@ -36,7 +36,8 @@ cluster in VM provided by Minishift.
        And stdout of command "minishift ssh -- "docker inspect --format={{.State.FinishedAt}} origin"" is not equal to "0001-01-01T00:00:00Z"
 
   Scenario: User deploys nodejs example application from OpenShift repository
-      When executing "oc new-app https://github.com/openshift/nodejs-ex -l name=myapp" succeeds
+     Given Minishift has state "Running"
+      When executing "oc new-app https://github.com/openshift/nodejs-ex -l name=myapp" retrying 10 times with wait period of 2 seconds
```
Comment (budhrg) [inline test/integration/features/cmd-openshift.feature]: :+1:

**Pull request merged**